datagrid_gtk3/db/sqlite.py

```diff
@@ -441,6 +447,62 @@ def _ensure_temp_view(self, cursor):
                 self.table.name, self.query
             ))
 
+    def _ensure_primary_key_column(self, conn):
+        """Ensure that we know what is the primary key.
+
+        :param conn: an open connection to the database
+        """
+        # FIXME: What to do when using temporary views?
+        if self.query:
+            return
+
+        with closing(conn.cursor()) as cursor:
+            cursor.execute('PRAGMA table_info(%s)' % (self.table.name, ))
+            rows = cursor.fetchall()
+
+        row_names = [row[1] for row in rows]
+
+        # First check if there's any row that matches self.ID_COLUMN
+        if self.ID_COLUMN in row_names:
+            return
+
+        # Then try to find any row that has its primary key flag set to True
+        for row in rows:
+            if row[5]:  # primary key
+                self.ID_COLUMN = row[1]
+                return
+
+        # If nothing worked and we need to ensure primary key or selected
+        # column, lets add it. Since sqlite doesn't allow us to add primary
+        # keys to existing tables, we are working around that by creating a
+        # new one with a primary key, copying everything from the previous
+        # table there.
+        if self._ensure_primary_key or self._ensure_selected_column:
+            t_name = self.table.name
+            candidate = '__id'
+            while candidate in row_names:
+                candidate = '_' + candidate
+
+            db = Database(self.db_file)
+            table = db.reflect(t_name)
+            new_cols = [
+                Column(c.name, c.type,
+                       primary_key=c.primary_key, default=c.default)
+                for c in table.columns.values()]
+            new_cols.append(Column(candidate, INTEGER, primary_key=True))
+            new_table = Table('__tmp', db.metadata, *new_cols)
```

Comment:
@hackedbellini What do you think about using '__tmp_' + `table_name` for a temporary table name? Just on the off chance such a generically named table as '__tmp' already exists...